### PR TITLE
Apply some lints of `clippy::pedantic` and `clippy::nursery`

### PR DESCRIPTION
**Pull request by sorairolake**
Apply the following:

- [`clippy::explicit_iter_loop`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#explicit_iter_loop)
- [`clippy::implicit_clone`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#implicit_clone)
- [`clippy::map_unwrap_or`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#map_unwrap_or)
- [`clippy::needless_for_each`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#needless_for_each)
- [`clippy::redundant_closure_for_method_calls`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#redundant_closure_for_method_calls)
- [`clippy::redundant_pub_crate`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#redundant_pub_crate)
- [`clippy::semicolon_if_nothing_returned`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#semicolon_if_nothing_returned)
- [`clippy::unreadable_literal`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#unreadable_literal)
- [`clippy::wildcard_imports`](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#wildcard_imports)

### CODE REVIEW

**Comment by blyxxyz** [comment]
I like most of these, but I'm not a fan of `clippy::use_self`.

File: src/main.rs [hunk 1]
```diff
@@ -422,9 +422,9 @@ fn run(args: Cli) -> Result<i32> {
 
         let mut request = request_builder.headers(headers).build()?;
 
-        headers_to_unset.iter().for_each(|h| {
+        for h in &headers_to_unset {
```
Comment (ducaale) [inline src/main.rs:425]: Can you replace the `h` variable here with `header`?

**Review by ducaale — COMMENTED** [review]
>I like most of these, but I'm not a fan of `clippy::use_self`.

I agree with @blyxxyz on this one. Let's revert the `structure name → Self` change. It would also be good to rename the PR's title to be more descriptive.

File: src/session.rs [hunk 2]
```diff
@@ -194,8 +194,8 @@ impl Session {
                     value: cookie.value().into(),
                     expires: cookie
                         .expires()
-                        .and_then(|v| v.datetime())
-                        .map(|v| v.unix_timestamp()),
+                        .and_then(cookie_crate::Expiration::datetime)
+                        .map(time::OffsetDateTime::unix_timestamp),
```
Comment (blyxxyz) [inline src/session.rs:198]: Doing this for getters feels weird.

Comment (sorairolake) [reply src/session.rs:198]: @blyxxyz Should I revert this part?

**Review by blyxxyz — APPROVED** [review]
Looks good!

**Pull request merged**